core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+      handler.allPerpendicularsSegmentArcBounded(
+        approximation, approximation.point0Ref, 0, approximation.point1Ref, 1, arc, true,
+      );
+    let maxPerp: CurveLocationDetailPair | undefined;
+    for (const perp of handler.grabPairedResults()) {
+      if (Geometry.isAlmostEqualEitherNumber(perp.detailA.fraction, 0, 1, Geometry.smallFraction))
+        continue; // rule out perpendiculars on arc ends
+      if (Geometry.isAlmostEqualEitherNumber(perp.detailB.fraction, 0, 1, Geometry.smallFraction))
+        continue; // rule out perpendiculars on approx ends
+      const error = perp.detailA.a = perp.detailB.a = perp.detailA.point.distanceXY(perp.detailB.point);
+      if (!maxPerp || maxPerp.detailA.a < error)
+        maxPerp = perp;
+    }
+    return maxPerp;
+  }
+};
+
+/**
+ * Intermediate class for computing the chain error of a sample-based approximation.
+ * @internal
+ */
+class ChainErrorProcessor extends ErrorProcessor {
+  private _maxPerpendicular: CurveLocationDetailPair | undefined;
+  protected constructor(ellipticalArc: Arc3d) {
+    super(ellipticalArc);
+    this._maxPerpendicular = undefined;
+  }
+  public get maxPerpendicular(): CurveLocationDetailPair | undefined {
+    return this._maxPerpendicular;
+  }
+  protected set maxPerpendicular(newMaxPerp: CurveLocationDetailPair) {
+    this._maxPerpendicular = newMaxPerp;
+  }
+  /**
+   * Update the chain approximation error for a given chain child that approximates the instance arc between the given
+   * fractions.
+   * * Fractional sweep of the arc is the smaller of the cyclic sweeps.
+   */
+  protected updateMaxPerpendicular(childApproximation: Arc3d | LineSegment3d, f0: number, f1: number): void {
+    const childPerp = ErrorProcessor.computePrimitiveErrorXY(
+      this.ellipticalArc.clonePartialCurve(f0, f1), childApproximation,
+    );
+    if (childPerp && (!this.maxPerpendicular || this.maxPerpendicular.detailA.a < childPerp.detailA.a))
+      this.maxPerpendicular = childPerp;
+  };
+
+};
+
+/**
+ * Processor for computing the error of a sample-based arc chain approximation.
+ * @internal
+ */
+class ArcChainErrorProcessor extends ChainErrorProcessor {
+  public constructor(ellipticalArc: Arc3d) {
+    super(ellipticalArc);
+  }
+  public override announceArc(arc: Arc3d, f0: number, f1: number): void {
+    this.updateMaxPerpendicular(arc, f0, f1);
+  }
+}
+
+/**
+ * Processor for computing the error of a sample-based linestring approximation.
+ * @internal
+ */
+class LineStringErrorProcessor extends ChainErrorProcessor {
+  public constructor(ellipticalArc: Arc3d) {
+    super(ellipticalArc);
+  }
+  public override announceChord(pt0: Point3d, pt1: Point3d, f0: number, f1: number): void {
+    this.updateMaxPerpendicular(LineSegment3d.create(pt0, pt1), f0, f1);
+  }
+}
+
+/**
+ * Intermediate processor class for computing samples for a subdivision-based approximation.
+ * @internal
+ */
+class SubdivisionErrorProcessor extends ErrorProcessor {
```

Comment:
It's not clear what this class is doing. What refinement set is, what is the count. Please add some comments to make it clear.

Reply:
I'll add more comments. Perhaps it will be clearer when you see how it's used below. The set is initially `q.fractions`. Then it is refined by adding a midpoint in each interval where the error of the announced approximant is too large.